cdap-app-fabric/src/main/java/co/cask/cdap/internal/provision/ProvisioningService.java

```diff
@@ -404,9 +420,17 @@ private Runnable createProvisionTask(ProvisioningTaskInfo taskInfo, Provisioner
   private Runnable createDeprovisionTask(ProvisioningTaskInfo taskInfo, Provisioner provisioner,
                                          Consumer<ProgramRunId> taskCleanup) {
     Map<String, String> properties = taskInfo.getProvisionerProperties();
-    ProvisionerContext context =
-      new DefaultProvisionerContext(taskInfo.getProgramRunId(), properties,
-                                    sparkCompat, createSSHContext(taskInfo.getSecureKeyInfo()));
+    ProvisionerContext context;
+    try {
+      context = createContext(taskInfo.getProgramRunId(), taskInfo.getUser(), properties,
```

Comment:
new to provisioning related changes, but when will this user be different from current user ?

Reply:
This is running in the CDAP master so it will be the cdap user. The user here is the user that the program will run as.